docs/UserGuide.md

```diff
@@ -108,6 +107,25 @@ Examples:
 * `deleteEvent 1` deletes the 1st event in the event list.
 * `deleteEvent 10` deletes the 10th event in the event list.
 
+### Edits an Event : `editEvent`
+
+Edits the event at the specified index with the specified attributes.
+
+Format: `editEvent EVENT_INDEX [n/EVENT_NAME] [l/LOCATION] [d/DATE] [t/TAG]`
+
+* The index refers to the index number shown in the displayed event list.
+* The index **must be a positive integer** that is within the range of the length of the member list.
+* At least one field to edit must be provided.
+* EVENT_NAME **must only contain** Alphanumeric Characters and spaces, and it should not be blank
+* LOCATION **must not** be blank and can take in any values.
+* DATE **must be a valid date** in the format of **YYYY-MM-DD** e.g. 2023-10-31.
+* TAG **must only contain** Alphanumeric Characters with no space in between.
+
+Examples:
+* `editEvent 5 n/Halloween Surprise Party l/UTR d/2023-10-31 t/sem1` edits the 5th event in the event list to change the name to Halloween
+Surprise Party, the location to UTR, the date to 31st October 2023 and the tag to sem1.
+* `editEvent 3 l/UCC Theater` edits the 3rd event in the event list to change the location to UCC Theater.
```

Comment:
Should we use code formatting here and stick to the same date format to make it clearer?
```suggestion
* `editEvent 5 n/Halloween Surprise Party l/UTR d/2023-10-31 t/sem1` edits the 5th event in the event list to change the name to `Halloween
Surprise Party`, the location to `UTR`, the date to `2023-10-31` and the tag to `sem1`.
* `editEvent 3 l/UCC Theater` edits the 3rd event in the event list to change the location to `UCC Theater`.
```